### PR TITLE
Zanzana: periodic sync of team members

### PR DESCRIPTION
**Pull request by kalleep**
**What is this feature?**
During transition period between current access control engine and openfga we want to do two things:
1. Write permissions in both stores
2. Create a reconciliation loop to ensure that the state in both stores matches

This is the initial work for the second point, and registers a reconciler for team memberships. On grafana side this is pretty easy at the moment, we just query for them all. Open fga does not offer an good api for this. The best I could come up with is to use the `Read` api. Sadly with current limitations we need to query the api for each resource and relationship that is possible.

So for team members that would be `number of teams * 2` because a user can either have member or admin relation ship. We can use this as a start but we need to figure out a better way to solve this because reconciliation will otherwise bombard the openfga api whenever we transition more resources to it and add it to the reconciliation loop.

I changed so that we are registering access control as a background service and trigger (for now) first `Sync` and then start the Reconciliation job. Run is called after `RegisterFixedRoles` so we should be good

**Which issue(s) does this PR fix?**:

Part of https://github.com/grafana/identity-access-team/issues/958

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: pkg/services/authz/zanzana/client/client.go [hunk 1]
```diff
@@ -94,6 +94,14 @@ func (c *Client) Check(ctx context.Context, in *openfgav1.CheckRequest) (*openfg
 	return c.client.Check(ctx, in)
 }
 
+func (c *Client) Read(ctx context.Context, in *openfgav1.ReadRequest) (*openfgav1.ReadResponse, error) {
+	ctx, span := tracer.Start(ctx, "authz.zanzana.client.Check")
```
Comment (alexanderzobnin) [inline pkg/services/authz/zanzana/client/client.go:98]:
```suggestion
	ctx, span := tracer.Start(ctx, "authz.zanzana.client.Read")
```

File: pkg/services/accesscontrol/dualwrite/reconciler.go [hunk 2]
```diff
@@ -88,6 +100,37 @@ func (z *ZanzanaSynchroniser) Sync(ctx context.Context) error {
...
+// Reconcile schedules as job that will run and reconcile resources between
+// legacy access control and zanzana.
+func (r *ZanzanaReconciler) Reconcile(ctx context.Context) error {
+	// Trigger full reconciliation initially
+	r.reconcile(ctx)
+
+	// FIXME:
+	// 1. We should be a bit graceful about reconciliations so we are not hammering dbs
+	// 2. We should be able to configure reconciliation interval
+	// 3. We should be able to detect if we already have a reconciliation loop running in case they take a long time
```
Comment (alexanderzobnin) [inline pkg/services/accesscontrol/dualwrite/reconciler.go:112]: I think we can add a flag `ZanzanaReconciler`, like `inProgress bool` and use mutex to set/release a flag. Then, `reconcile()` func can check if there's any other reconcile in progress and return immediately.

Comment (kalleep) [reply pkg/services/accesscontrol/dualwrite/reconciler.go:112]: Yeah or just an atomic would do fine

Comment (kalleep) [reply pkg/services/accesscontrol/dualwrite/reconciler.go:112]: oh and we need to handle multiple instances so we should probably use serverlock

**Review by alexanderzobnin — APPROVED** [review]

**Pull request merged**